Mimir.MongoDB/Repositories/ProductRepository.cs

```diff
@@ -40,4 +55,124 @@ public async Task<ProductDocument> GetByProductIdAsync(Guid productId)
 
         return productDocument;
     }
-}
+
+    public class ProductFilter
+    {
+        public ProductType? ProductType { get; set; }
+        public ItemType? ItemType { get; set; }
+        public ItemSubType? ItemSubType { get; set; }
+        public ProductSortBy? SortBy { get; set; }
+        public Enums.SortDirection? SortDirection { get; set; } = Enums.SortDirection.Ascending;
+    }
+
+    private static FilterDefinition<ProductDocument> BuildFilter(ProductFilter? productFilter)
+    {
+        var filterBuilder = Builders<ProductDocument>.Filter;
+        var filter = filterBuilder.Empty;
+
+        if (productFilter?.ProductType is not null)
+        {
+            filter &= filterBuilder.Eq(x => x.Object.ProductType, productFilter.ProductType);
+        }
+
+        if (productFilter?.ItemType is not null)
+        {
+            filter &= filterBuilder.Eq("Object.TradableItem.ItemType", productFilter.ItemType.ToString());
+        }
+
+        if (productFilter?.ItemSubType is not null)
+        {
+            filter &= filterBuilder.Eq("Object.TradableItem.ItemSubType", productFilter.ItemSubType.ToString());
+        }
+
+        return filter;
+    }
+
+    private IExecutable<ProductDocument> ApplySorting(ProductFilter productFilter, FilterDefinition<ProductDocument> filter,
+        IFindFluent<ProductDocument, ProductDocument> find)
+    {
+        productFilter.SortDirection ??= Enums.SortDirection.Ascending;
+
+        switch (productFilter.SortBy)
+        {
+            case ProductSortBy.Cp:
+                break;
+            case ProductSortBy.Crystal:
+                break;
```

Comment:
I would recommend removing sort criteria from the `ProductSortBy` enumeration that are not yet supported: `Cp`, `Crystal`